core/storage/beam_job/gae_models.py

```diff
@@ -0,0 +1,169 @@
+    # This state may only be set via a Cloud Dataflow jobs.update call, and only
+    # if the job has not yet reached another terminal state.
+    'CANCELLED',
+    # The job is in the process of draining. A draining job has stopped pulling
+    # from its input sources and is processing any data that remains in-flight.
+    # This state may be set via a Cloud Dataflow jobs.update call, but only as a
+    # transition from RUNNING. Jobs that are draining may only transition to
+    # DRAINED, CANCELLED, or FAILED.
+    'DRAINING',
+    # The job has been drained. A drained job terminated by stopping pulling
+    # from its input sources and processing any data that remained in-flight
+    # when draining was requested. This state is a terminal state, may only be
+    # set by the Cloud Dataflow service, and only as a transition from DRAINING.
+    'DRAINED',
+    # The job was successfully updated, meaning that this job was stopped and
+    # another job was started, inheriting state from this one. This is a
+    # terminal job state. This state may only be set by the Cloud Dataflow
+    # service, and only as a transition from RUNNING.
+    'UPDATED',
+    # The job has successfully completed. This is a terminal job state. This
+    # state may be set by the Cloud Dataflow service, as a transition from
+    # RUNNING. It may also be set via a Cloud Dataflow jobs.update call, if the
+    # job has not yet reached a terminal state.
+    'DONE',
+    # The job has has failed. This is a terminal job state. This state may only
+    # be set by the Cloud Dataflow service, and only as a transition from
+    # RUNNING.
+    'FAILED',
+    # The job's run state isn't specified.
+    'UNKNOWN')
+
+
+class BeamJobRunModel(base_models.BaseModel):
+    """Represents the execution of an individual Apache Beam job.
+
+    IDs are generated by Google Cloud Dataflow, not Oppia, and are guaranteed to
+    be unique:
+    https://cloud.google.com/dataflow/docs/reference/rest/v1b3/projects.jobs#resource:-job
+    """
+
+    # The name of the job class that implements the job's logic.
+    job_name = datastore_services.StringProperty(indexed=True, required=True)
+    # The arguments provided to the job run.
+    job_arguments = (
+        datastore_services.StringProperty(indexed=True, repeated=True))
+    # The state of the job at the time the model was last updated.
+    latest_job_state = datastore_services.StringProperty(indexed=True, choices=[
+        BeamJobState.RUNNING.value,
+        BeamJobState.PENDING.value,
+        BeamJobState.STOPPED.value,
+        BeamJobState.CANCELLING.value,
+        BeamJobState.CANCELLED.value,
+        BeamJobState.DRAINING.value,
+        BeamJobState.DRAINED.value,
+        BeamJobState.UPDATED.value,
+        BeamJobState.DONE.value,
+        BeamJobState.FAILED.value,
+        BeamJobState.UNKNOWN.value,
+    ], required=True)
+
+    @property
+    def in_terminal_state(self):
```

Comment:
In general, we add such functionalities in the domain/object layer. (i.e, not in the model layer!) I'm fine with this, just wanted to check in case if it's fine to move t to the domain_class*

Reply:
Sure, I'll do that in the next PR.